tests/src/test/java/alluxio/proxy/s3/S3ClientRestApiTest.java

```diff
@@ -205,6 +209,173 @@ public void deleteNonEmptyBucket() throws Exception {
+    // Verify the directory is created for the new bucket.
+    Assert.assertTrue(mFileSystemMaster.listStatus(uri, ListStatusOptions.defaults()).isEmpty());
+
+    // Prepare a bucket with two objects:
+    // - /file1
+    // - /file2
+    AlluxioURI file1 = new AlluxioURI(uri.getPath() + "/file1");
+    mFileSystemMaster.createFile(file1, CreateFileOptions.defaults());
+    AlluxioURI file2 = new AlluxioURI(uri.getPath() + "/file2");
+    mFileSystemMaster.createFile(file2, CreateFileOptions.defaults());
+
+    // Expected result, with max-keys = 1.
+    List<URIStatus> objectsList = new ArrayList<>();
+    objectsList.add(
+        new URIStatus(mFileSystemMaster.getFileInfo(file1, GetStatusOptions.defaults())));
+    objectsList.add(
+        new URIStatus(mFileSystemMaster.getFileInfo(file2, GetStatusOptions.defaults())));
+    String maxKeys = "1";
+    String continuationToken = file1.getPath();
+    ListBucketResult expected = new ListBucketResult(
+        AlluxioURI.SEPARATOR + bucket, objectsList,
+        ListBucketOptions.defaults().setMaxKeys("1").setContinuationToken(continuationToken));
+
+    // Verify
+    HashMap<String, String> maxKeysParam = new HashMap<>();
+    maxKeysParam.put("max-keys", maxKeys);
+    maxKeysParam.put("continuation-token", continuationToken);
+    new TestCase(mHostname, mPort, S3_SERVICE_PREFIX + AlluxioURI.SEPARATOR + bucket, maxKeysParam,
+        HttpMethod.GET, expected,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE)).run();
+  }
+
+  @Test
+  public void listEmptyBucket() throws Exception {
+    final String bucket = "empty-bucket-to-list";
+    AlluxioURI uri = new AlluxioURI(AlluxioURI.SEPARATOR + bucket + AlluxioURI.SEPARATOR);
+    new TestCase(mHostname, mPort, S3_SERVICE_PREFIX + AlluxioURI.SEPARATOR + bucket, NO_PARAMS,
+        HttpMethod.PUT, null, TestCaseOptions.defaults()).run();
+    // Verify the directory is created for the new bucket.
+    Assert.assertTrue(mFileSystemMaster.listStatus(uri, ListStatusOptions.defaults()).isEmpty());
+
+    List<URIStatus> listStatusResult = new ArrayList<>();
+    ListBucketResult expected = new ListBucketResult(
+        AlluxioURI.SEPARATOR + bucket, listStatusResult, ListBucketOptions.defaults());
+
+    new TestCase(mHostname, mPort, S3_SERVICE_PREFIX + AlluxioURI.SEPARATOR + bucket, NO_PARAMS,
+        HttpMethod.GET, expected,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE))
+        .run();
+  }
+
+  @Test
+  public void getNonExistingBucket() throws Exception {
+    final String bucketName = "non-existing-bucket";
+
+    try {
+      // Delete a non-existing bucket should fail.
+      new TestCase(mHostname, mPort, S3_SERVICE_PREFIX + AlluxioURI.SEPARATOR + bucketName,
+          NO_PARAMS, HttpMethod.GET, null,
+          TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE)).run();
+      Assert.fail("get a non-existing bucket should fail");
```

Comment:
this will throw AssertionError too, then if the test fails, there is no way to detect the error

Reply:
done, good catch, thanks